eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/ManagementInputLibrary.cs

```diff
@@ -166,5 +196,36 @@ ResourceMetadata BuildResourceMetadata(InputDecoratorInfo decorator)
                     parentResource);
             }
         }
+
+        internal bool IsResourceUpdateModel(InputModelType model)
+        {
+            if (_resourceUpdateModels?.Contains(model) == true)
+            {
+                return true;
+            }
+
+            const string ResourceUpdateModelId = KnownManagementTypes.ResourceUpdateModelId;
+
+            var currentModel = model;
+            while (currentModel != null)
+            {
+                if (currentModel.CrossLanguageDefinitionId.Equals(ResourceUpdateModelId, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+                currentModel = currentModel.BaseModel;
+            }
+
+            return false;
+        }
+
+        internal string FindEnclosingResourceNameForResourceUpdateModel(InputModelType model)
+        {
+            if (_resourceUpdateModelToResourceNameMap?.TryGetValue(model, out var resourceName) == true)
+            {
+                return resourceName;
+            }
+            throw new InvalidOperationException($"Could not find enclosing resource name for resource update model '{model.Name}'");
```

Comment:
We should try to use the emitter diagnostics rather than throwing if this is something that users might actually hit.

Reply:
e.g. https://github.com/Azure/azure-sdk-for-net/blob/main/eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/CollectionResultDefinition.cs#L109-L112

Reply:
Thanks for pointing out. In this case, it's an internal bug rather than something that users should hit, So I think we should stop the code generation process by throwing an exception.

Reply:
This should never happen, we can even safely remove this throw, just leave it here for code completion.